modules/conversion-center/classes/render/core-render.php

```diff
@@ -0,0 +1,217 @@
+		$has_bio_title       = ! empty( $bio_title_value );
+		$has_ctas            = ! empty( $ctas_value );
+		$has_icons           = ! empty( $icons_value );
+		$has_identity_image  = ! empty( $identity_image_value ) && ( ! empty( $identity_image_value['url'] || ! empty( $identity_image_value['id'] ) ) );
+
+		// Render conditions
+		if ( ! $has_bio_heading ) {
+			return;
+		}
+		?>
+		<div class="e-link-in-bio__content-container">
+			<?php if ( $has_identity_image ) :
+				$widget->add_render_attribute( 'identity', [
+					'class' => [ 'e-link-in-bio__identity', "e-link-in-bio__identity--{$identity_image_props_style}" ],
+				] );
+				?>
+				<figure <?php echo esc_attr( $widget->get_render_attribute_string( 'identity' ) ); ?>>
+					<?php if ( ! empty( $identity_image_value['id'] ) ) {
+						echo wp_get_attachment_image( $identity_image_value['id'], 'medium', false, [
+							'class' => 'e-link-in-bio__identity-image',
+						] );
+					} else {
+						$widget->add_render_attribute( 'identity_image', [
+							'alt'   => '',
+							'class' => 'e-link-in-bio__identity-image',
+							'src'   => esc_url( $identity_image_value['url'] ),
+						] );
+						?>
+						<img <?php echo esc_attr( $widget->get_render_attribute_string( 'icon-link' ) ); ?> />
+					<?php }; ?>
+				</figure>
+			<?php endif; ?>
+			<?php if ( $has_bio_heading || $has_bio_title || $has_bio_description ) : ?>
+				<div class="e-link-in-bio__content">
+					<?php if ( $has_bio_heading ) {
+						$widget->add_render_attribute( 'heading', 'class', 'e-link-in-bio__heading' );
+						$bio_heading_output = sprintf( '<%1$s %2$s>%3$s</%1$s>', Utils::validate_html_tag( $bio_heading_props_tag ), $widget->get_render_attribute_string( 'heading' ), $bio_heading_value );
+						Utils::print_unescaped_internal_string( $bio_heading_output );
+					} ?>
+					<?php if ( $has_bio_title ) {
+						$widget->add_render_attribute( 'title', 'class', 'e-link-in-bio__title' );
+						$bio_title_output = sprintf( '<%1$s %2$s>%3$s</%1$s>', Utils::validate_html_tag( $bio_title_props_tag ), $widget->get_render_attribute_string( 'title' ), $bio_title_value );
+						Utils::print_unescaped_internal_string( $bio_title_output );
+					} ?>
+					<?php if ( $has_bio_description ) {
+						$widget->add_render_attribute( 'description', 'class', 'e-link-in-bio__description' );
+						$bio_description_output = sprintf( '<p %1$s>%2$s</p>', $widget->get_render_attribute_string( 'description' ), $bio_description_value );
+						Utils::print_unescaped_internal_string( $bio_description_output );
+					} ?>
+				</div>
+			<?php endif; ?>
+			<?php if ( $has_icons ) : ?>
+				<div class="e-link-in-bio__icons">
+					<?php
+					foreach ( $icons_value as $key => $icon ) {
+						// Bail if no icon
+						if ( empty( $icon['icon_icon'] ) ) {
+							break;
+						}
+
+						// Check for link format based on type
+						$formatted_link = $icon['icon_url']['url'] ?? '';
```

Comment:
```suggestion
						$formatted_link = $this->get_formatted_link();
```
Maybe extract this as well.